crates/voicevox_core_python_api/src/lib.rs

```diff
@@ -361,6 +371,75 @@ impl Synthesizer {
     }
 }
 
+#[pyclass]
+#[derive(Debug, Clone)]
+struct UserDict {
+    dict: voicevox_core::UserDict,
+}
+
+#[pymethods]
+impl UserDict {
+    #[new]
+    fn new() -> Self {
+        Self {
+            dict: voicevox_core::UserDict::new(),
+        }
+    }
+
+    fn load(&mut self, path: &str) -> PyResult<()> {
+        self.dict.load(path).into_py_result()?;
+        Ok(())
+    }
+
+    fn save(&self, path: &str) -> PyResult<()> {
+        self.dict.save(path).into_py_result()?;
+        Ok(())
+    }
+
+    fn add_word(
+        &mut self,
+        #[pyo3(from_py_with = "to_rust_user_dict_word")] word: UserDictWord,
+        py: Python,
+    ) -> PyResult<PyObject> {
+        let uuid = self.dict.add_word(word).into_py_result()?;
+
+        to_py_uuid(py, uuid)
+    }
+
+    fn update_word(
+        &mut self,
+        #[pyo3(from_py_with = "to_rust_uuid")] word_uuid: Uuid,
+        #[pyo3(from_py_with = "to_rust_user_dict_word")] word: UserDictWord,
+    ) -> PyResult<()> {
+        self.dict.update_word(word_uuid, word).into_py_result()?;
+        Ok(())
+    }
+
+    fn remove_word(
+        &mut self,
+        #[pyo3(from_py_with = "to_rust_uuid")] word_uuid: Uuid,
+    ) -> PyResult<()> {
+        self.dict.remove_word(word_uuid).into_py_result()?;
+        Ok(())
+    }
+
+    fn import_dict(&mut self, other: &UserDict) -> PyResult<()> {
+        self.dict.import(&other.dict).into_py_result()?;
+        Ok(())
+    }
+
+    #[getter]
+    fn words<'py>(&self, py: Python<'py>) -> PyResult<&'py PyDict> {
+        let dict = PyDict::new(py);
+        for (uuid, word) in self.dict.words() {
+            let uuid = to_py_uuid(py, uuid.to_owned())?;
+            let word: UserDictWord = word.clone();
+            dict.set_item(uuid, to_py_user_dict_word(py, &word)?)?;
+        }
+        Ok(dict)
```

Comment:
この辺は好みの問題ですし、清く正しいPython使いやGo使いの人であればfor文で書くと思うのですが、私ならこう書きます。

```suggestion
        use pyo3::types::IntoPyDict as _;

        let words = self
            .dict
            .words()
            .iter()
            .map(|(&uuid, word)| {
                let uuid = to_py_uuid(py, uuid)?;
                let word = to_py_user_dict_word(py, word)?;
                Ok((uuid, word))
            })
            .collect::<PyResult<Vec<_>>>()?;
        Ok(words.into_py_dict(py))
```